tardis/transport/montecarlo/weighted_packet_source.py

```diff
@@ -0,0 +1,177 @@
+import numexpr as ne
+import numpy as np
+from astropy import constants as const
+from astropy import units as u
+
+from tardis.transport.montecarlo.packet_source import (
+    BasePacketSource,
+    HDFWriterMixin,
+)
+from tardis.util.base import intensity_black_body
+
+
+class BlackBodyWeightedSource(BasePacketSource, HDFWriterMixin):
+    """
+    Simple packet source that generates Blackbody packets for the Montecarlo
+    part.
+
+    Parameters
+    ----------
+    radius : astropy.units.Quantity
+        Initial packet radius
+    temperature : astropy.units.Quantity
+        Absolute Temperature.
+    base_seed : int
+        Base Seed for random number generator
+    legacy_secondary_seed : int
+        Secondary seed for global numpy rng (Deprecated: Legacy reasons only)
+    """
+
+    hdf_properties = ["radius", "temperature", "base_seed"]
+    hdf_name = "black_body_weighted_source"
+
+    @classmethod
+    def from_simulation_state(cls, simulation_state, *args, **kwargs):
+        return cls(
+            simulation_state.r_inner[0],
+            simulation_state.t_inner,
+            *args,
+            **kwargs,
+        )
+
+    def __init__(self, radius=None, temperature=None, **kwargs):
+        self.radius = radius
+        self.temperature = temperature
+        super().__init__(**kwargs)
+
+    def create_packets(self, no_of_packets, *args, **kwargs):
+        if self.radius is None or self.temperature is None:
+            raise ValueError("Black body Radius or Temperature isn't set")
+        return super().create_packets(no_of_packets, *args, **kwargs)
+
+    def create_packet_radii(self, no_of_packets):
+        """
+        Create packet radii
+
+        Parameters
+        ----------
+        no_of_packets : int
+            number of packets to be created
+
+        Returns
+        -------
+        Radii for packets
+            numpy.ndarray
+        """
+        return np.ones(no_of_packets) * self.radius.cgs
+
+    def create_packet_nus(self, no_of_packets, l_samples=1000):
```

Comment:
the documentation of this function is wrong

Reply:
I've new fixed it